test/utils/wasm3_engine.cpp

```diff
@@ -74,14 +74,18 @@ fizzy::bytes_view Wasm3Engine::get_memory() const
     return {data, size};
 }
 
-void Wasm3Engine::set_memory(fizzy::bytes_view memory)
+bool Wasm3Engine::set_memory(fizzy::bytes_view memory)
 {
-    if (memory.empty())
-        return;
-
     uint32_t size;
     const auto data = m3_GetMemory(m_runtime, &size, 0);
-    std::memcpy(data, memory.data(), size);
+    if (data == nullptr)
+        return memory.empty();
+
+    if (memory.data() == nullptr)
```

Comment:
Why `data() == nullptr` here, while `empty()` elsewhere?

Also not sure if `empty()` means to zero out memory, to be honest.

Reply:
`memory.data() == nullptr` is supposed to check whether it was called like `set_memory(bytes_view{})`. But it can be `memory.empty()`, too.

`empty()` doesn't zero put (it's a view anyway, can't be zeroed out), it checks whether it's empty.
(BTW both `bytes_view{}` and `bytes_view(ptr, 0)` are empty, but second one has non-null `data()`)